### PR TITLE
Add unique uuid to loop resources for pipeline versioning

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #

**Description of your changes:**
Due to the limitation of the Tekton taskSpec API, currently we need to define the loop resources in the namespace scope. However, in order to do versioning with loop resources, we will need to create an unique uuid for each version which is removed in the kfp 1.4.0 release [commit](https://github.com/kubeflow/pipelines/commit/2f1db59798d1a4c49e5cba7ec60644eca3fe5175#diff-5fc12b53e14acb39e9be93fc5b91cc87d41a72643a9faab0b054e7f60cdf9d51)

Thus, we need to bring back the generated uuid feature for kfp-tekton until Tekton supports taskSpec for custom task.

/cc @fenglixa @ckadner 

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

**Comment by Tomcli** [comment]
/hold
there's a bug with the name matching in the nested loops use case

**Comment by Tomcli** [comment]
@fenglixa @ckadner This PR is ready to review.

File: sdk/python/kfp_tekton/compiler/_tekton_hander.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def _handle_tekton_pipeline_variables(pipeline_run):
     return pipeline_run
 
 
-def _handle_tekton_custom_task(custom_task: dict, workflow: dict):
```
Comment (ckadner) [inline sdk/python/kfp_tekton/compiler/_tekton_hander.py:59]: can we use this opportunity to rename the file from 
`_tekton_hander.py` to 
`_tekton_handler.py` (with the "L") or
`_tekton_task_handler.py`

Comment (Tomcli) [reply sdk/python/kfp_tekton/compiler/_tekton_hander.py:59]: done

**Comment by ckadner** [comment]
@fenglixa -- this looks good to me. Do you have any comments?

**Comment by fenglixa** [comment]
Yes, It's good to me. Thanks @Tomcli

File: sdk/python/kfp_tekton/compiler/_tekton_hander.py (hunk 1 above)
Comment (ckadner) [reply sdk/python/kfp_tekton/compiler/_tekton_hander.py:59]: Hey Tommy, please update the import statement in `compiler.py` as well:

```
  File "/home/travis/build/kubeflow/kfp-tekton/sdk/python/kfp_tekton/compiler/compiler.py", line 45, in <module>

    from kfp_tekton.compiler._tekton_hander import _handle_tekton_pipeline_variables, _handle_tekton_custom_task

ModuleNotFoundError: No module named 'kfp_tekton.compiler._tekton_hander'
```

**Comment by ckadner** [comment]
@Tomcli -- seems like the "Golden" YAML files need to be regenerated

```Bash
make unit_test GENERATE_GOLDEN_YAML=True
```

**Comment by ckadner** [comment]
Thanks Tommy

/lgtm

**Pull request merged**